dash/include/dash/GlobRef.h

```diff
@@ -145,7 +145,7 @@ class GlobRef
   /**
    * Move Assignment: Redirects to Copy Assignment
    */
-  self_t& operator=(self_t&& other) {
+  self_t& operator=(self_t&& other) noexcept {
```

Comment:
In other places you use `DASH_NOEXCEPT`. Is there a difference?

Reply:
There is a difference. `DASH_NOEXCEPT` is only `noexcept` if we are building in release mode. I implemented it for `GlobIter` now and we should refactor this in a separate PR to be consistent in all classes.